ckanext/fairdatapoint/tests/test_record_provider.py

```diff
@@ -105,3 +105,27 @@ def test_orcid_call(self, mocker):
             expected = Graph().parse(Path(TEST_DATA_DIRECTORY, "Project_27866022694497978_out.ttl")).serialize()
             assert mock.called
             assert actual == expected
+
+    @pytest.mark.parametrize("harvest_catalogs, expected_results", [
+        (False, {
+            'catalog=http://example.com/catalog1;dataset=http://example.com/dataset1': URIRef(
+                'http://example.com/dataset1')
+        }),
+        (True, {
+            'catalog=http://example.com/catalog1;dataset=http://example.com/dataset1': URIRef(
+                'http://example.com/dataset1'),
+            'catalog=http://example.com/catalog1': URIRef('http://example.com/catalog1')
+        })
+    ])
+    def test_process_catalogs(self, mocker, harvest_catalogs, expected_results):
+        fdp_get_graph = mocker.MagicMock(name="get_data")
+        mocker.patch("ckanext.fairdatapoint.harvesters.domain.fair_data_point.FairDataPoint.get_graph",
+                     new=fdp_get_graph)
+        fdp_get_graph.return_value = Graph().parse(Path(TEST_DATA_DIRECTORY, "process_catalogs.ttl"))
+
+        self.fdp_record_provider.harvest_catalogs = harvest_catalogs
+        actual_result = self.fdp_record_provider._process_catalog("http://example.com/catalog1")
+
+        # Assertions
+        for key, value in expected_results.items():
+            assert actual_result[key] == value
```

Comment:
Do we need to check, if there are no additional entries in actual_result ?

Reply:
Could do with the == operator if we don't want additional entries.

Reply:
aangepast